api_core/google/api_core/protobuf_helpers.py

```diff
@@ -76,3 +82,150 @@ def get_messages(module):
         if inspect.isclass(candidate) and issubclass(candidate, Message):
             answer[name] = candidate
     return answer
+
+
+def _resolve_subkeys(key, separator='.'):
+    """Given a key which may actually be a nested key, return the top level
+    key and any nested subkeys as separate values.
+
+    Args:
+        key (str): A string that may or may not contain the separator.
+        separator (str): The namespace separator. Defaults to `.`.
+
+    Returns:
+        Tuple[str, str]: The key and subkey(s).
+    """
+    subkey = None
+    if separator in key:
+        index = key.index(separator)
+        subkey = key[index + 1:]
+        key = key[:index]
+    return key, subkey
+
+
+def get(msg_or_dict, key, default=_SENTINEL):
+    """Retrieve a key's value from a protobuf Message or dictionary.
+
+    Args:
+        mdg_or_dict (Union[~google.protobuf.message.Message, Mapping]): the
+            object.
+        key (str): The key to retrieve from the object.
+        default (Any): If the key is not present on the object, and a default
+            is set, returns that default instead. A type-appropriate falsy
+            default is generally recommended, as protobuf messages almost
+            always have default values for unset values and it is not always
+            possible to tell the difference between a falsy value and an
+            unset one. If no default is set then raises :class:`KeyError` will
```

Comment:
ISTM you want "raises" deleted in this sentence?

Reply:
Done.